crnk-core/src/main/java/io/crnk/core/engine/internal/dispatcher/controller/ResourceDeleteController.java

```diff
@@ -1,43 +1,58 @@
 package io.crnk.core.engine.internal.dispatcher.controller;
 
-import java.io.Serializable;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-
 import io.crnk.core.engine.dispatcher.Response;
 import io.crnk.core.engine.document.Document;
+import io.crnk.core.engine.document.Resource;
 import io.crnk.core.engine.http.HttpMethod;
+import io.crnk.core.engine.information.resource.ResourceInformation;
 import io.crnk.core.engine.internal.dispatcher.path.JsonPath;
 import io.crnk.core.engine.internal.dispatcher.path.ResourcePath;
 import io.crnk.core.engine.internal.repository.ResourceRepositoryAdapter;
 import io.crnk.core.engine.query.QueryAdapter;
 import io.crnk.core.engine.registry.RegistryEntry;
 import io.crnk.core.engine.result.Result;
 import io.crnk.core.engine.result.ResultFactory;
+import io.crnk.core.exception.RepositoryNotFoundException;
+import io.crnk.core.exception.RequestBodyException;
+import io.crnk.core.exception.RequestBodyNotFoundException;
 import io.crnk.core.repository.response.JsonApiResponse;
 
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
 public class ResourceDeleteController extends BaseController {
 
 	@Override
 	public boolean isAcceptable(JsonPath jsonPath, String method) {
-		return !jsonPath.isCollection()
-				&& jsonPath instanceof ResourcePath
+		return jsonPath instanceof ResourcePath
 				&& HttpMethod.DELETE.name().equals(method);
 	}
 
 	@Override
-	public Result<Response> handleAsync(JsonPath jsonPath, QueryAdapter queryAdapter, Document requestBody) {
+	public Result<Response> handleAsync(JsonPath jsonPath, QueryAdapter queryAdapter, Document requestDocument) {
 		RegistryEntry registryEntry = jsonPath.getRootEntry();
 		Collection<Serializable> ids = jsonPath.getIds();
 		logger.debug("using registry entry {}", registryEntry);
 		logger.debug("deleting ids={}", ids);
-
 		List<Result<JsonApiResponse>> results = new ArrayList<>();
-		for (Serializable id : ids) {
+		if (ids == null) {
+			List<Resource> resourceBodies = getRequestBodys(requestDocument, jsonPath, HttpMethod.DELETE);
 			ResourceRepositoryAdapter resourceRepository = registryEntry.getResourceRepository();
-			Result<JsonApiResponse> result = resourceRepository.delete(id, queryAdapter);
+			ids = resourceBodies.stream()
+					.map(resource -> getParsedId(resource, resourceRepository.getRepositoryInformation().getResource()))
+					.collect(Collectors.toList());
+			Result<JsonApiResponse> result = resourceRepository.delete(ids, queryAdapter);
 			results.add(result);
+		} else {
+			for (Serializable id : ids) {
```

Comment:
maybe the else could be aligned with the delete(ids, queryAdapter) just above. Not sure...